File: ios/NutritionReportService.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(NutritionReportService, NSObject)

RCT_EXTERN_METHOD(generateNutritionPDF:(NSString *)foodRecord withUserDetails: (NSString *)userDetails withDuration: (float)duration
				  withResolver:(RCTPromiseResolveBlock)resolve
				  withRejecter:(RCTPromiseRejectBlock)reject)

@end

File: ios/NutritionReportService.swift
@objc(NutritionReportService)
class NutritionReportService: NSObject {
    
    @objc(generateNutritionPDF:withUserDetails:withDuration:withResolver:withRejecter:)
    func generateNutritionPDF(
        foodRecord: String,
        userDetails: String,
        duration: Float,
        resolve:@escaping RCTPromiseResolveBlock,
        reject:@escaping RCTPromiseRejectBlock
    ) -> Void {
        
        reject("", "error1.localizedDescription", Error.self as? Error)
    }
    
    
}
